alerts/worker.py

```diff
@@ -0,0 +1,51 @@
+"""Long-running worker that executes alerting jobs."""
+import logging
+import threading
+
+from alerts.jobs import AiringShowsJob
+
+ONE_DAY = 60 * 60 * 24
+
+
+class AlertWorker(threading.Thread):
+    """Check for airing jobs every day.
+
+    Notes
+    -----
+    Controlling the update time of day is not yet supported.
+    The alerting job will be executed every day at the time
+    that this thread is started.
+    """
+
+    period_seconds = ONE_DAY
+    _logger = logging.getLogger('alerts')
+
+    @staticmethod
+    def main_thread_is_alive() -> bool:
+        """Check if the main thread is still running.
+
+        This check is required because the web server may not be able
+        to correctly stop this thread and the Timer it has fired.
+        """
+        for thread in threading.enumerate():
+            if thread.getName().find('MainThread') != -1:
+                return thread.is_alive()
+        return False
+
+    def schedule_next(self):
+        """Schedule another run of this job after the configured period."""
+        return threading.Timer(self.period_seconds, lambda: self.run())
+
```

Comment:
So the job for the next day will run 24h after the start of this day's job or after the end of this day's job ? if its the end, then it will not always be at the same hour and will be later everyday...

Reply:
No because the timer doesn't wait for the job to end before being relaunched.

Reply:
ok perfect then